include/vara/Solver/ConfigurationFactory.h

```diff
@@ -0,0 +1,133 @@
+#ifndef VARA_FEATURE_CONFIGURATIONFACTORY_H
+#define VARA_FEATURE_CONFIGURATIONFACTORY_H
+
+#include "vara/Configuration/Configuration.h"
+#include "vara/Feature/FeatureModel.h"
+#include "vara/Sampling/SamplingMethods.h"
+#include "vara/Solver/Solver.h"
+#include "vara/Solver/SolverFactory.h"
+
+#include <vector>
+
+namespace vara::solver {
+
+/// This class represents a configuration iterator that lazily traverses
+/// over all configurations.
+class ConfigurationIterator {
+public:
+  class Iterator {
+  public:
```

Comment:
```suggestion
  public:
      using iterator_category = ?;
      using value_type = ?;
      using difference_type = ?;
      using pointer = ?;
      using reference = ?;
```
Should define iterator traits to be easy usable with standard algorithms

Reply:
Will do that in the next revision.